### PR TITLE
Remove remaining c-style casts from the spec where appropriate.

### LINKED ISSUE #379: Find all C-style casts in specification and replace them with `static_cast`s
...except the necessary C-style cast in the implementation of <i>`c-style-cast`</i>.

### PR DESCRIPTION
**Pull request by griwes**
Resolves #379.

### CODE REVIEW

File: std_execution.bs [hunk 1]
```diff
@@ -4577,21 +4577,21 @@ template &lt;class E>  // arguments are not associated entities ([lib.tmpl-heads
 
         where <code><i>connect-awaitable-promise</i></code> is the promise type of <code><i>connect-awaitable</i></code>, and where <code><i>connect-awaitable</i></code> suspends at the <i>initial suspends point</i> ([dcl.fct.def.coroutine]), and:
 
-          1. <i>set-value-expr</i> first evaluates `co_await (S&&) s`, then suspends the coroutine and evaluates `execution::set_value((R&&) r)` if <code><i>await-result-type</i>&lt;S, <i>connect-awaitable-promise</i>></code> is <code><i>cv</i> void</code>; otherwise, it evaluates `auto&& res = co_await (S&&) s`, then suspends the coroutine and evaluates `execution::set_value((R&&) r, (decltype(res)) res)`.
+          1. <i>set-value-expr</i> first evaluates `co_await std::forward<S>(s)`, then suspends the coroutine and evaluates `execution::set_value(std::forward<R>(r))` if <code><i>await-result-type</i>&lt;S, <i>connect-awaitable-promise</i>></code> is <code><i>cv</i> void</code>; otherwise, it evaluates `auto&& res = co_await std::forward<S>(s)`, then suspends the coroutine and evaluates `execution::set_value(std::forward<R>(r), std::forward<decltype(res)>(res))`.
```
Comment (ericniebler) [inline std_execution.bs:4580]: Can't tell from my phone, but if memory serves these should be moves, not forwards. I could be totally wrong though.

File: std_execution.bs [hunk 2]
```diff
@@ -4577,21 +4577,21 @@ template &lt;class E>  // arguments are not associated entities ([lib.tmpl-heads
...
 
             If the call to `execution::set_value` exits with an exception, the coroutine is resumed and the exception is immediately propagated in the context of the coroutine.
 
             [<i>Note</i>: If the call to `execution::set_value` exits normally, then the <code><i>connect-awaitable</i></code> coroutine is never resumed. --<i>end note</i>]
 
-          2. <i>set-error-expr</i> first suspends the coroutine and then executes `execution::set_error((R&&) r, std::move(ep))`.
+          2. <i>set-error-expr</i> first suspends the coroutine and then executes `execution::set_error(std::forward<R>(r), std::move(ep))`.
```
Comment (ericniebler) [inline std_execution.bs:4586]:
```suggestion
          2. <i>set-error-expr</i> first suspends the coroutine and then executes `execution::set_error(std::move(r), std::move(ep))`.
```

File: std_execution.bs [hunk 3]
```diff
@@ -4577,21 +4577,21 @@ template &lt;class E>  // arguments are not associated entities ([lib.tmpl-heads
...
 
             [<i>Note</i>: The <code><i>connect-awaitable</i></code> coroutine is never resumed after the call to `execution::set_error`. --<i>end note</i>]
 
           3. <code><i>operation-state-task</i></code> is a type that models `operation_state`. Its `execution::start` resumes the <code><i>connect-awaitable</i></code> coroutine, advancing it past the initial suspend point.
 
           4. Let `p` be an lvalue reference to the promise of the <code><i>connect-awaitable</i></code> coroutine, let `b` be a `const` lvalue reference to the receiver `r`. Then `tag_invoke(tag, p, as...)` is expression-equivalent to `tag(b, as...)` for any set of arguments `as...` and for any `tag` whose type satisfies <code><i>forwarding-receiver-query</i></code>.
 
-          5. The expression `p.unhandled_stopped()` is expression-equivalent to `(execution::set_stopped((R&&) r), noop_coroutine())`.
+          5. The expression `p.unhandled_stopped()` is expression-equivalent to `(execution::set_stopped(std::forward<R>(r), noop_coroutine())`.
```
Comment (ericniebler) [inline std_execution.bs:4594]:
```suggestion
          5. The expression `p.unhandled_stopped()` is expression-equivalent to `(execution::set_stopped(std::move(r)), noop_coroutine())`.
```

**Review by ericniebler — CHANGES_REQUESTED** [review]

File: std_execution.bs (hunk 1 above)
Comment (griwes) [reply std_execution.bs:4580]: Ah yes, because this is within a function that accepts them with decayed types. Will fix.

File: std_execution.bs [hunk 4]
```diff
@@ -4577,21 +4577,21 @@ template &lt;class E>  // arguments are not associated entities ([lib.tmpl-heads
...
+          2. <i>set-error-expr</i> first suspends the coroutine and then executes `execution::set_error(std::move(r), std::move(ep))`.
 
             [<i>Note</i>: The <code><i>connect-awaitable</i></code> coroutine is never resumed after the call to `execution::set_error`. --<i>end note</i>]
 
           3. <code><i>operation-state-task</i></code> is a type that models `operation_state`. Its `execution::start` resumes the <code><i>connect-awaitable</i></code> coroutine, advancing it past the initial suspend point.
 
           4. Let `p` be an lvalue reference to the promise of the <code><i>connect-awaitable</i></code> coroutine, let `b` be a `const` lvalue reference to the receiver `r`. Then `tag_invoke(tag, p, as...)` is expression-equivalent to `tag(b, as...)` for any set of arguments `as...` and for any `tag` whose type satisfies <code><i>forwarding-receiver-query</i></code>.
 
-          5. The expression `p.unhandled_stopped()` is expression-equivalent to `(execution::set_stopped((R&&) r), noop_coroutine())`.
+          5. The expression `p.unhandled_stopped()` is expression-equivalent to `(execution::set_stopped(std::move(r), noop_coroutine())`.
```
Comment (ericniebler) [inline std_execution.bs:4594]:
```suggestion
          5. The expression `p.unhandled_stopped()` is expression-equivalent to `(execution::set_stopped(std::move(r)), noop_coroutine())`.
```

**Review by ericniebler — APPROVED** [review]

**Pull request merged**